File: solutions/059/059.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func main() {
	bs, _ := ioutil.ReadFile("resources/cipher.txt")
	cipher := strings.Split(strings.Replace(string(bs), "\n", "", -1), ",")

	size := len(cipher)

	sum := 0
	key := [3]int{'g', 'o', 'd'}

	for i := 0; i < size; i++ {
		x, _ := strconv.Atoi(cipher[i])
		sum += x ^ (key[i%3])
	}

	fmt.Println(sum)
}

File: solutions/059/059.py
from operator import xor

with open("./resources/cipher.txt") as f:
    txt = [int(i) for i in f.read().strip().split(',')]

s = 0
key = 'god'
for i, x in enumerate(txt):
    s += xor(x, ord(key[i % 3]))

print(s)
